crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -793,6 +799,11 @@ impl<'a> Resolver<'a> {
                 }
             }
             Type::MutableReference(element) => Self::find_numeric_generics_in_type(element, found),
+            Type::FmtString(_, fields) => {
+                for field in fields {
+                    Self::find_numeric_generics_in_type(field, found);
+                }
+            }
```

Comment:
Both FmtString and String have numeric generics as their length fields so we should add the same check done on arrays to both of them (its an omission that str doesn't already have this check). We should also recur on an array's element since nested arrays will be possible in the future.

Reply:
Makes sense. Done